sdk-extensions/incubator/src/main/java/io/opentelemetry/sdk/extension/incubator/fileconfig/ConfigurationReader.java

```diff
@@ -8,25 +8,104 @@
 import com.fasterxml.jackson.annotation.JsonSetter;
 import com.fasterxml.jackson.annotation.Nulls;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import io.opentelemetry.sdk.autoconfigure.spi.ConfigurationException;
 import io.opentelemetry.sdk.extension.incubator.fileconfig.internal.model.OpenTelemetryConfiguration;
+import java.io.BufferedReader;
+import java.io.IOException;
 import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import java.util.regex.MatchResult;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 import org.snakeyaml.engine.v2.api.Load;
 import org.snakeyaml.engine.v2.api.LoadSettings;
 
 final class ConfigurationReader {
 
-  private static final ObjectMapper MAPPER =
-      new ObjectMapper()
-          // Create empty object instances for keys which are present but have null values
-          .setDefaultSetterInfo(JsonSetter.Value.forValueNulls(Nulls.AS_EMPTY));
+  private static final Pattern ENV_VARIABLE_REFERENCE =
+      Pattern.compile("\\$\\{env:([a-zA-Z_]+[a-zA-Z0-9_]*)}");
```

Comment:
As it reads now, `${ env:MY_VAR }` would fail the parse due to the space after/before the curly braces. I think other systems are pretty tolerant of whitespace inside the `${ ... }` variable marker.  Bash isn't, but python and javascript are.

I'm guessing this isn't clarified in the spec yet? I wouldn't hold up this PR for this, but it would be good to give it some consideration. Even `"\\$\\{\\s*env:([a-zA-Z_]+[a-zA-Z0-9_]*)\\s*}"` might be enough?

Reply:
I think we start strict and relax later if we need to.

I'm planning on opening a spec PR for this - just working out a few things about how the substitution would work with JSON, which various maintainers insisted on supporting with file configuration.